src/apps/protocols/ftp/mode/filescount.pm

```diff
@@ -121,36 +121,38 @@ sub countFiles {
             my $hash = pop @$list;
             my $dir = $hash->{name};
             my $level = $hash->{level};
-                        
+
             if (!(@files = apps::protocols::ftp::lib::ftp::execute($self, command => $map_commands{ls}->{$self->{ssl_or_not}}->{name}, command_args => [$dir]))) {
                 # Cannot list we skip
                 next;
             }
-
+            # this loop is recursive, when we find a directory we add it to the list used by the for loop.
+            # max_depth is used to limit the depth we search.
+            # same behaviour as cifs(samba) and sftp protocol.
             foreach my $line (@files) {
                 # IIS: 05-13-15  10:59AM              1184403 test.jpg
                 next if ($line !~ /(\S+)\s+\S+\s+\S+\s+\S+\s+\S+\s+\S+\s+\S+\s+\S+\s+(.*)/ &&
-                         $line !~ /^\s*\S+\s*\S+\s*(\S+)\s+(.*)/);
+                    $line !~ /^\s*\S+\s*\S+\s*(\S+)\s+(.*)/);
                 my ($rights, $filename) = ($1, $2);
                 my $bname = basename($filename);
                 next if ($bname eq '.' || $bname eq '..');
                 my $name = $dir . '/' . $bname;
-                
-                if (defined($self->{option_results}->{filter_file}) && $self->{option_results}->{filter_file} ne '' &&
-                    $name !~ /$self->{option_results}->{filter_file}/) {
-                    $self->{output}->output_add(long_msg => sprintf("Skipping '%s'", $name));
-                    next;
-                }
-            
+
                 if ($rights =~ /^(d|<DIR>)/i) {
-                    if (defined($self->{option_results}->{max_depth}) && $level + 1 <= $self->{option_results}->{max_depth}) {
-                        push @$list, { name => $name, level => $level + 1};
+                    # need 2 if or directory not matching the max_depth condition would be treated as files
```

Comment:
Your comment is hard to understand and will be even harder once the context if this change is lost.